File: ebean-externalmapping-xml/src/test/java/org/tests/model/basic/Order.java
package org.tests.model.basic;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OrderBy;
import javax.persistence.Table;
import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "o_order")
public class Order extends BasicDomain {

  public enum Status {
    NEW,
    APPROVED,
    SHIPPED,
    COMPLETE
  }

  @Enumerated(value = EnumType.ORDINAL)
  Status status = Status.NEW;
  LocalDate orderDate;
  LocalDate shipDate;

  @ManyToOne(cascade = CascadeType.PERSIST)
  Customer customer;

  @OneToMany(cascade = CascadeType.ALL, mappedBy = "order")
  @OrderBy("id asc, orderQty asc, cretime desc")
  List<OrderDetail> details;

}

File: tests/test-kotlin/src/test/kotlin/org/example/order/Order.kt
package org.example.order

import javax.persistence.CascadeType
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.persistence.Table
import javax.persistence.Version

@Entity
@Table(name = "t_order")
class Order(
  customer: Customer,
) {
  @Id
  val id: Long = 0

  @ManyToOne
  val customer: Customer = customer

  @OneToMany(mappedBy = "order", cascade = [CascadeType.ALL], orphanRemoval = true)
  val items: MutableList<OrderItem> = mutableListOf()

  @Version
  val version: Long = 0
}
